test/e2e/common/node/container_probe.go

```diff
@@ -1036,27 +1036,26 @@ func runReadinessFailTest(ctx context.Context, f *framework.Framework, pod *v1.P
 }
 
 func gRPCServerPodSpec(readinessProbe, livenessProbe *v1.Probe, containerName string) *v1.Pod {
-	etcdLocalhostAddress := "127.0.0.1"
+	agnhostLocalhostAddress := "127.0.0.1"
 	if framework.TestContext.ClusterIsIPv6() {
-		etcdLocalhostAddress = "::1"
+		agnhostLocalhostAddress = "::1"
 	}
-	etcdURL := fmt.Sprintf("http://%s", net.JoinHostPort(etcdLocalhostAddress, "2379"))
+	agnhostURL := fmt.Sprintf("http://%s", net.JoinHostPort(agnhostLocalhostAddress, "6379"))
 	return &v1.Pod{
 		ObjectMeta: metav1.ObjectMeta{Name: "test-grpc-" + string(uuid.NewUUID())},
 		Spec: v1.PodSpec{
 			Containers: []v1.Container{
 				{
 					Name:  containerName,
-					Image: imageutils.GetE2EImage(imageutils.Etcd),
+					Image: imageutils.GetE2EImage(imageutils.Agnhost),
 					Command: []string{
-						"/usr/local/bin/etcd",
+						"/usr/local/bin/agnhost",
```

Comment:
I don't think this will work. Have you tried it?

Reply:
in this article you can find out how the arguments would look like for the agnhost: https://kubernetes.io/blog/2022/05/13/grpc-probes-now-in-beta/

Reply:
@SergeyKanzhelev 
Thanks for your review! I fixed the code so please check it!